src/main/scala/edu/ie3/simona/service/primary/PrimaryServiceWorker.scala

```diff
@@ -348,6 +377,22 @@ case object PrimaryServiceWorker {
       timePattern: String
   ) extends InitPrimaryServiceStateData
 
+  /** Specific implementation of [[InitPrimaryServiceStateData]], if the source
+    * to use utilizes an SQL database.
+    *
+    * @param sqlParams
+    *   Parameters regarding SQL connection and table selection
+    * @param timeSeriesUuid
+    *   Unique identifier of the time series to read
+    * @param simulationStart
+    *   Wall clock time of the beginning of simulation time
+    */
+  final case class SqlInitPrimaryServiceStateData(
+      sqlParams: SqlParams,
```

Comment:
A matter of taste, but I don't like handing over container classes aka. data structures.

Reply:
Well the sql params contain five fields, which would blow up this constructor. So I think it makes sense to use the config class here.